ql/cashflows/couponpricer.hpp

```diff
@@ -49,6 +49,9 @@ namespace QuantLib {
         //@{
         virtual Real swapletPrice() const = 0;
         virtual Rate swapletRate() const = 0;
+        virtual Rate swapletRate(const Date& date) const { 
+            return swapletRate();
+        }
```

Comment:
I'm not sure that this is the interface that makes most sense (there's no coupon rate defined until the end of the coupon, only the accrued amount) and thinking as a maintainer, once we publish it, it gets very hard to change as it needs to go through deprecation etc.

Since the overnight coupon is the only one that uses it right now, what would you say about not declaring the method here, defining in the overnight pricer only, and using a cast in `OvernightIndexedCoupon::accruedAmount` to access it?  I would avoid changes in `FloatingRateCoupon` too.

Reply:
I think the latest commit should be close to what you've described, I understand you're maintenance concern on previous solution.